test/DotNetty.Transport.Tests.Performance/Libuv/TcpPerfSpecs.cs

```diff
@@ -0,0 +1,144 @@
+
+            this.message = Encoding.UTF8.GetBytes("ABC");
+
+            this.inboundThroughputCounter = context.GetCounter(InboundThroughputCounterName);
+            this.outboundThroughputCounter = context.GetCounter(OutboundThroughputCounterName);
+            var counterHandler = new CounterHandlerInbound(this.inboundThroughputCounter);
+            this.signal = new ManualResetEventSlimReadFinishedSignal(this.resetEvent);
+
+            // reserve up to 10mb of 16kb buffers on both client and server; we're only sending about 700k worth of messages
+            this.serverBufferAllocator = new PooledByteBufferAllocator();
+            this.clientBufferAllocator = new PooledByteBufferAllocator();
+
+            ServerBootstrap sb = new ServerBootstrap()
+                .Group(this.serverGroup)
+                .Channel<TcpServerChannel>()
+                .ChildOption(ChannelOption.Allocator, this.serverBufferAllocator)
+                .ChildHandler(new ActionChannelInitializer<TcpChannel>(channel =>
+                {
+                    channel.Pipeline
+                        .AddLast(GetEncoder())
+                        .AddLast(GetDecoder())
+                        .AddLast(counterHandler)
+                        .AddLast(new ReadFinishedHandler(this.signal, WriteCount));
+                }));
+
+            Bootstrap cb = new Bootstrap()
+                .Group(this.clientGroup)
+                .Channel<TcpChannel>()
+                .Option(ChannelOption.Allocator, this.clientBufferAllocator)
+                .Handler(new ActionChannelInitializer<TcpChannel>(
+                    channel =>
+                    {
+                        channel.Pipeline
+                            .AddLast(GetEncoder())
+                            .AddLast(GetDecoder())
+                            .AddLast(new CounterHandlerOutbound(this.outboundThroughputCounter));
+                    }));
+
+            // start server
+            this.serverChannel = sb.BindAsync(TestAddress).Result;
+
+            // connect to server
+            this.clientChannel = cb.ConnectAsync(this.serverChannel.LocalAddress).Result;
+        }
+
+        [PerfBenchmark(Description = "Measures how quickly and with how much GC overhead a TcpChannel --> TcpServerChannel connection can decode / encode realistic messages, 10 writes per flush",
+            NumberOfIterations = IterationCount, RunMode = RunMode.Iterations)]
+        [CounterMeasurement(InboundThroughputCounterName)]
+        [CounterMeasurement(OutboundThroughputCounterName)]
+        [GcMeasurement(GcMetric.TotalCollections, GcGeneration.AllGc)]
+        [MemoryMeasurement(MemoryMetric.TotalBytesAllocated)]
+        public void TcpDuplexThroughput10MessagesPerFlush(BenchmarkContext context)
+        {
+            this.clientChannel.EventLoop.Execute(() =>
+            {
+                for (int i = 0; i < WriteCount; i++)
```

Comment:
can we run from within the loop here? E.g.
[https://github.com/Azure/DotNetty/blob/dev/test/DotNetty.Transport.Tests.Performance/Sockets/TcpSocketChannelPerfSpec.cs#L135](https://github.com/Azure/DotNetty/blob/dev/test/DotNetty.Transport.Tests.Performance/Sockets/TcpSocketChannelPerfSpec.cs#L135)
it avoids cross-thread work posting which is not what we want reflected here.